src/Services/DatabaseBackup/MysqlCustomDatabaseBackup.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Liip/FunctionalTestBundle
+ *
+ * (c) Lukas Kahwe Smith <smith@pooteeweet.org>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Liip\FunctionalTestBundle\Services\DatabaseBackup;
+
+use Doctrine\Common\DataFixtures\Executor\AbstractExecutor;
+
+/**
+ * @author Aleksey Tupichenkov <alekseytupichenkov@gmail.com>
+ *
+ * It's class created just for example that how to create database backup/restore service
+ */
+class MysqlCustomDatabaseBackup extends AbstractDatabaseBackup
+{
+    public function getBackupName(): string
+    {
+        return $this->container->getParameter('kernel.cache_dir').'/test_mysql_'.md5(serialize($this->metadatas).serialize($this->classNames)).'.sql';
+    }
+
+    public function isBackupActual(): bool
+    {
+        $backupDBFileName = $this->getBackupName();
+        $backupReferenceFileName = $backupDBFileName.'.ser';
+
+        return file_exists($backupDBFileName) && file_exists($backupReferenceFileName) && $this->isBackupUpToDate($backupDBFileName);
+    }
+
+    public function backup(AbstractExecutor $executor): void
+    {
+        $params = $this->connection->getParams();
+        if (isset($params['master'])) {
+            $params = $params['master'];
+        }
+
+        $dbName = isset($params['dbname']) ? $params['dbname'] : '';
+        $dbHost = isset($params['host']) ? $params['host'] : '';
+        $dbPort = isset($params['port']) ? $params['port'] : '';
+        $dbUser = isset($params['user']) ? $params['user'] : '';
+        $dbPass = isset($params['password']) ? $params['password'] : '';
+
+        $executor->getReferenceRepository()->save($this->getBackupName());
+
+        exec("mysqldump -h $dbHost -u $dbUser -p$dbPass $dbName > {$this->getBackupName()}");
```

Comment:
If the MySQL database is in a Docker container, this command won't work?

Reply:
I use docker too, it will works fine if you install mysql-client in container with symfony app :)